src/mod/applications/mod_http_cache/mod_http_cache.c

```diff
@@ -1176,6 +1178,181 @@ static switch_status_t http_get(url_cache_t *cache, http_profile_t *profile, cac
 		goto done;
 	}
 
+done:
+
+	if (headers) {
+		switch_curl_slist_free_all(headers);
+	}
+
+	return status;
+}
+/**
+ * Deletes a remote file via HTTP
+ * @param cache the cache
+ * @param profile the HTTP profile
+ * @param url The cached URL entry
+ * @param session the (optional) session
+ * @return SWITCH_STATUS_SUCCESS if successful
+ */
+static switch_status_t http_delete(url_cache_t *cache, http_profile_t *profile, cached_url_t *url, switch_core_session_t *session)
+{
+	switch_status_t status = SWITCH_STATUS_SUCCESS;
+	switch_curl_slist_t *headers = NULL;  /* optional linked-list of HTTP headers */
+	switch_CURL *curl_handle = NULL;
+	http_get_data_t del_data = {0};
+	long httpRes = 0;
+	switch_CURLcode curl_status = CURLE_UNKNOWN_OPTION;
+
+	/* set up HTTP DELETE */
+	del_data.fd = 0;
+	del_data.url = url;
+
+	/* find profile for domain */
+	if (!profile) {
+		profile = url_cache_http_profile_find_by_fqdn(cache, url->url);
+	}
+
+	if (profile && profile->append_headers_ptr) {
+		headers = profile->append_headers_ptr(profile, headers, "DELETE", 0, "", url->url, 0, NULL);
+	}
+
+	curl_handle = switch_curl_easy_init();
+	switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(session), SWITCH_LOG_DEBUG, "opening %s for URL cache headers\n", del_data.url->filename);
+	switch_curl_easy_setopt(curl_handle, CURLOPT_FOLLOWLOCATION, 1);
+	switch_curl_easy_setopt(curl_handle, CURLOPT_MAXREDIRS, 10);
+	switch_curl_easy_setopt(curl_handle, CURLOPT_FAILONERROR, 1);
+	switch_curl_easy_setopt(curl_handle, CURLOPT_NOSIGNAL, 1);
+	if (headers) {
+		switch_curl_easy_setopt(curl_handle, CURLOPT_HTTPHEADER, headers);
+	}
+	switch_curl_easy_setopt(curl_handle, CURLOPT_URL, del_data.url->url);
+	switch_curl_easy_setopt(curl_handle, CURLOPT_CUSTOMREQUEST, "DELETE");
+	switch_curl_easy_setopt(curl_handle, CURLOPT_WRITEFUNCTION, get_file_callback);
+	switch_curl_easy_setopt(curl_handle, CURLOPT_WRITEDATA, (void *) &del_data);
+	switch_curl_easy_setopt(curl_handle, CURLOPT_HEADERFUNCTION, get_header_callback);
+	switch_curl_easy_setopt(curl_handle, CURLOPT_WRITEHEADER, (void *) url);
+	switch_curl_easy_setopt(curl_handle, CURLOPT_USERAGENT, "freeswitch-http-cache/1.0");
+	if (cache->connect_timeout > 0) {
+		switch_curl_easy_setopt(curl_handle, CURLOPT_CONNECTTIMEOUT, cache->connect_timeout);
+	}
+	if (cache->download_timeout > 0) {
+		switch_curl_easy_setopt(curl_handle, CURLOPT_TIMEOUT, cache->download_timeout);
+	}
+	if (!cache->ssl_verifypeer) {
+		switch_curl_easy_setopt(curl_handle, CURLOPT_SSL_VERIFYPEER, 0L);
+	} else {
+		/* this is the file with all the trusted certificate authorities */
+		if (!zstr(cache->ssl_cacert)) {
+			switch_curl_easy_setopt(curl_handle, CURLOPT_CAINFO, cache->ssl_cacert);
+		}
+	}
```

Comment:
I would move these (and the one in the other function) to a seperate function that actually use the same common setopt instead of duplicating them everytime.  Reduce the risk of future edit of 1 of them without changing them all.

Reply:
@Marc O Chouinard Thanks for your quick feedback. I know that I am duplicating code and that it would be a good idea to put them into a common function, but I was not sure if I should use the common function in the existing functions http_put and http_get because I am very hesitant to change existing code that I also didn't write. Additionally I am not sure whether that function should be inside the http_cache module or whether it it should be generic to be used with mod_httapi. 
Lastly I am not sure the preferred way to make a function like that. Would it be better to pass in a reference/pointer and return a switch_status or would it be better to have it return a CURL struct? 
I'll try and make updates shortly

Reply:
I would do it for everything that is identical, like the CURLOPT_USERAGENT for example.  And all the one that are not common, leave them separate for now unless you have a good curl option knowledge.  It would at lest reduce current duplicate code, make sure functionality remain the same, and provide a framework for the next person who come along to change it who have more knowledge about curl to consolidate further.  For now, this is stuff hard coded for mod_http_cache, so it fine to leave it internal to that module.
I'm not sure what you mean with the result status.  Currently these return nothing, so you wouldn't have to deal with it, or just return Success by default for now.  What I would do is send a pointer of the curl_handle and maybe also the cache struct if it was used in all the functions and any other value that is standards and commonly needed, and the common function would access that pointer directly.

Reply:
Thanks that hits the nail on the head of what I was asking. I made some updates according to your suggestions.